python/ray/util/scheduling_strategies.py

```diff
@@ -72,9 +72,95 @@ def __init__(
         self._fail_on_unavailable = _fail_on_unavailable
 
 
+def _validate_values(values):
+    if values is None or len(values) == 0:
+        raise ValueError("The value of label cannot be empty.")
+
+    index = 0
+    for value in values:
+        if not isinstance(value, str):
+            raise ValueError(
+                f"Expected value {str(value)} in position {index} to be of type str"
+            )
+        index = index + 1
+
+
+class In:
+    def __init__(self, *values):
+        _validate_values(values)
+        self.values = list(values)
+
+
+class NotIn:
+    def __init__(self, *values):
+        _validate_values(values)
+        self.values = list(values)
+
+
+class Exists:
+    def __init__(self):
+        pass
+
+
+class DoesNotExist:
+    def __init__(self):
+        pass
+
+
+class _LabelMatchExpression:
+    """An expression used to select node by node's labels
+    Attributes:
+        key: the key of label
+        operator: In、NotIn、Exists、DoesNotExist
+    """
+
+    def __init__(self, key: str, operator: Union[In, NotIn, Exists, DoesNotExist]):
+        self.key = key
+        self.operator = operator
+
+
+LabelMatchExpressionsT = Dict[str, Union[In, NotIn, Exists, DoesNotExist]]
+
+
+@PublicAPI(stability="alpha")
+class NodeLabelSchedulingStrategy:
+    """Label based node affinity scheduling strategy
+
+    scheduling_strategy=NodeLabelSchedulingStrategy({
+          "region": In("us"),
+          "gpu_type": Exists()
+    })
+    """
+
+    def __init__(
+        self, hard: LabelMatchExpressionsT, *, soft: LabelMatchExpressionsT = None
+    ):
+        self.hard = _convert_map_to_expressions(hard, "hard")
+        self.soft = _convert_map_to_expressions(soft, "soft")
+
+
+def _convert_map_to_expressions(map_expressions: LabelMatchExpressionsT, param: str):
+    expressions = []
+    if map_expressions is None:
+        return expressions
+
+    for key, value in map_expressions.items():
```

Comment:
We should check the map_expressions is of type map, the error message can be:

The {param} parameter must be a map (e.g. {"key1": In("value1")}) but got type {type(map_expressions)}

We should also check the key is of type string, the error message can be:

The map key of the {param} parameter must be str (e.g. {"key1": In("value1")}) but got {key} of type {type(key)}

Reply:
Also add test for the invalid cases

Reply:
Fixed.